core/src/main/java/jenkins/security/ConfidentialStore.java

```diff
@@ -122,17 +122,28 @@ void clear() {
 
         @Override
         protected void store(ConfidentialKey key, byte[] payload) throws IOException {
-            LOGGER.fine(() -> "storing " + key.getId() + " " + Util.getDigestOf(Util.toHexString(payload)));
+            //called only from tests, get hex string of sha 256 for logging payload
+            logPayload("storing", key.getId(), payload);
```

Comment:
the original code would have been more concise here.

`LOGGER.fine(() -> "storing " + key.getId() + " " + Util.getHexOfSHA256DigestOf(payload)));` 

As this is a supplier form the String concatination is not be called unless fine level logging has been enabled.